### ISSUE TITLE
AWS RDS: COPY to a file is not permitted

### ISSUE THREAD

**Issue by ZipBrandon** [denishpatel/pg-clone-schema#102]
Invoking with `select clone_schema('sourceschema', 'sample_clone', 'DATA');` I am unable to copy data.  `select clone_schema('sourceschema', 'sample_clones', 'NODATA');` will succeed though.
```
Query 1 ERROR: ERROR:  Version: 1.15  December 22, 2022  Action: Copy Rows  Diagnostics: line=SQL statement "COPY sourceschema.profiles_appuser TO '/tmp/cloneschema.tmp' WITH DELIMITER AS ',';"
PL/pgSQL function clone_schema(text,text,cloneparms[]) line 2028 at EXECUTE. 0A000. COPY to a file is not supported
CONTEXT:  PL/pgSQL function clone_schema(text,text,cloneparms[]) line 2115 at RAISE
```

**Comment by MichaelDBA** [comment]
Yes, PG DBaaS won't currently work for tables that have user-defined datatypes that are defined in the source schema.  If the datatypes are in the public schema it will work as you saw with the **citext** datatype that you had defined in the public schema.  This problem does not exist for **onprem** PG instances since clone_schema gets around this problem by writing the data to disk on the server, and then reading it back in to the target tables bypassing the schema qualifications for data types.

As a workaround, defined you user-defined datatypes in the public schema like you did for citext.

**Comment by ZipBrandon** [comment]
I don't think this due to user-defined datatypes but possibly that RDS doesn't have access to a `/tmp` file storage.

**Comment by MichaelDBA** [comment]
That's exactly what I mean.  Sorry if I'm not being clear.  clone_schema does a COPY to and from the server file system to get around this problem and since you are in the cloud, this attempt fails since clone_schema does not have access to the file system.
